src/main/java/school/lemon/changerequest/java/introduction/hw2/Task4.java

```diff
@@ -2,6 +2,26 @@
 
 public class Task4 {
     public static String charAt(String s, int i) {
-        return "";
+
+
+        if (i < 0 && Math.abs(i) <= s.length()) {
+            StringBuilder s1 = new StringBuilder(s);
+            s1.reverse();
+            i = Math.abs(i) - 1;
+            return String.valueOf(s1.charAt(i));
+        }
+
+
+        if (i < 0 && Math.abs(i) > s.length()) {
+            StringBuilder s1 = new StringBuilder(s);
+            s1.reverse();
+            i = s.length() - (Math.abs(i) - 1);
+            return String.valueOf(s1.charAt(i));
+        }
+
+        if (i > 0 && i >= s.length()) {
+            i = i - s.length();
+        }
+        return String.valueOf(s.charAt(i));
```

Comment:
`%` - is modulus operator in Java.
For example for positive index, result index on which you can get needed char could be calculated in following way:
`int index %=s.length()`.